### DISCUSSION TITLE
extractVersion without bringing in patch version

### DISCUSSION THREAD

**Discussion by gregoryduckworth** [renovatebot/renovate#8277] [Request Help] [answered]
I have gradle which I am trying to update to only provide `major`.`minor` versions however when attempting to use the `dataSourceTemplate` as `github-releases`  (in a `regexManager`) with the following configuration:
``` "extractVersion": "^v(?<version>\\d+\\.\\d+\\.\\d+)?"``` it shows I can get the correct version but ``` "extractVersion": "^v(?<version>\\d+\\.\\d+)(?:\\.\\d+)?"``` fails and it cannot get the releases from Github. I feel like I've misunderstood how to use `extractVersion` correctly. 

Can anyone point me in the right direction?

Note: configuration in question.
```
"packageRules": {
      "packageNames": ["gradle"],
      "extractVersion": "^v(?<version>\\d+\\.\\d+)(?:\\.\\d+)?"
  },
  "regexManagers": [
    {
      "fileMatch": ["Dockerfile*$"],
      "matchStrings": ["GRADLE_VERSION=(?<currentValue>.*)\n"],
      "depNameTemplate": "gradle",
      "lookupNameTemplate": "gradle/gradle",
      "datasourceTemplate": "github-releases"
    }
  ]
```

**Comment by viceice** [comment, marked as answer]
Use `gradle-version` datasource.  Checkout how we use it 🙃 

* here: https://github.com/renovatebot/docker-gradle/blob/master/Dockerfile#L10-L11
* https://github.com/renovatebot/presets-internal/blob/master/docker-images.json#L37-L43

**Reply by viceice** [reply]
It will update `6.2.1` => `6.3` => `6.3.1` 😉

**Reply by gregoryduckworth** [reply]
Thanks for the links, looks like I was missing `"versioningTemplate": "gradle"`

My final config for anyone else was:
```
"packageRules": {
      "packageNames": ["gradle"],
      "updateTypes": ["major", "minor"]
  },
  "regexManagers": [
    {
      "fileMatch": ["Dockerfile*$"],
      "matchStrings": ["GRADLE_VERSION=(?<currentValue>.*)\n"],
      "depNameTemplate": "gradle",
      "datasourceTemplate": "gradle-version",
      "versioningTemplate": "gradle"
    }
  ]
```
